From: Simon McVittie <smcv@debian.org>
Date: Mon, 28 Jan 2019 12:46:56 +0000
Subject: make-test-*: Don't sandbox the icon validator

During build-time tests, bwrap doesn't necessarily work. In particular,
official Debian autobuilders can't enter namespaces.

We continue to leave the sandbox enabled in the build-export calls in
tests/test-extensions.sh, tests/test-unsigned-summaries.sh
and tests/test-update-remote-configuration.sh, which are already
skipped if bwrap isn't available. This means we exercise both the
normal and --disable-sandbox code paths.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/make-test-app.sh     | 2 +-
 tests/make-test-runtime.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/make-test-app.sh b/tests/make-test-app.sh
index 33108df..86afd48 100755
--- a/tests/make-test-app.sh
+++ b/tests/make-test-app.sh
@@ -96,7 +96,7 @@ ln -s -t ${DIR}/files/share/locale ../../share/runtime/locale/fr/share/fr
 
 flatpak build-finish --command=hello.sh ${DIR}
 mkdir -p repos
-flatpak build-export ${collection_args} ${GPGARGS-} ${EXPORT_ARGS-} ${REPO} ${DIR}
+flatpak build-export --disable-sandbox ${collection_args} ${GPGARGS-} ${EXPORT_ARGS-} ${REPO} ${DIR}
 rm -rf ${DIR}
 
 # build a locale extension
diff --git a/tests/make-test-runtime.sh b/tests/make-test-runtime.sh
index eabda7b..67c9e33 100755
--- a/tests/make-test-runtime.sh
+++ b/tests/make-test-runtime.sh
@@ -85,5 +85,5 @@ else
 fi
 
 mkdir -p repos
-flatpak build-export ${collection_args} --runtime ${GPGARGS-} ${REPO} ${DIR}
+flatpak build-export ${collection_args} --disable-sandbox --runtime ${GPGARGS-} ${REPO} ${DIR}
 rm -rf ${DIR}
